liwords-ui/src/chat/chat.tsx

```diff
@@ -50,43 +85,142 @@ export const Chat = React.memo((props: Props) => {
     [props.presences]
   );
 
-  // Chat auto-scrolls when the last entity is visible.
-  const [hasUnreadChat, setHasUnreadChat] = useState(false);
-  const chatTab = selectedChatTab === 'CHAT' ? tabContainerElement : null;
-  const [chatAutoScroll, setChatAutoScroll] = useState(true);
   const doChatAutoScroll = useCallback(() => {
     if (chatAutoScroll && chatTab) {
-      const desiredScrollTop = chatTab.scrollHeight - chatTab.clientHeight;
       // Doing this conditionally may help browser performance.
       // (Not sure.)
-      if (chatTab.scrollTop !== desiredScrollTop) {
-        chatTab.scrollTop = desiredScrollTop;
+      if (chatTab.scrollTop !== chatTab.scrollHeight) {
+        chatTab.scrollTop = chatTab.scrollHeight;
       }
     }
   }, [chatAutoScroll, chatTab]);
 
+  const fetchChannels = useCallback(() => {
+    axios
+      .post(
+        toAPIUrl('user_service.SocializeService', 'GetActiveChatChannels'),
+        {
+          number: 20,
+          offset: 0,
+        },
+        { withCredentials: true }
+      )
+      .then((res) => {
+        const newChannels: ActiveChatChannels.AsObject = {
+          channelsList:
+            res.data.channels.map((ch: any) => {
```

Comment:
We should try to avoid doing the `any`. I know why you are -- it's this protobuf to object stuff and the extremely annoying fact that it sends `display_name` when you invoke it with JSON but the `AsObject` defines it as `displayName`. Maybe there's no real way around this yet. What I've been doing is just defining shadow types that are very similar but with the other type of casing. I have to figure out a better way to do this.

Reply:
I'll do that